### PR TITLE
Better Travis CI

### PR DESCRIPTION
**Pull request by rdoursenaud**
<pre>
    NEW:
    Cleaned up routines for better readability of both declaration and results.
    PHP versions now really covered.
        The old code forced install of PHP and didn't use Travis provided versions.
        This resulted in the process not being executed with the declared PHP version.
    Dropped MySQL in favor of MariaDB.
        This is now the FLOSS community standard.
        This should help avoid problems with buggy MySQL releases.
    Fast finish enabled to show results faster.
    Optimized tools installation with composer.
        The right version of the tool is installed for the PHP version under test.
    New PHP linter to check for syntax errors.
        Parallelized for better speed.
    Apache + PHP FPM for testing webservices.
        The previous mod_php configuration was not supported on Travis.
    New global DEBUG environment variable to show verbose output with configuration files content.
    IRC notification on #dolibarr@freenode for community awareness.
    
    FIXES:
    Bug in scripts preventing execution with environmentalized PHP.
    Wrong detection of MAIN_URL_ROOT under specific circumstances.
        $_SERVER["DOCUMENT_ROOT"] empty and $_SERVER["SCRIPT_NAME"] populated.
    Relative ignore directive in coding style ruleset to avoid bypassing test.
    Unit test errors without an exit status.
        This prevented the CI from properly detecting and reporting the error.
    
    TODOS:
    PostgreSQL support.
        This one is tricky since we only have a MySQL dump and the syntax is not directly compatible.
    SQLite support.
        Disabled in core at the moment.
    Nginx + PHP FPM support.
        Test webservices on the second most popular webserver.
    Run dev/* checks.
        We have a nice collection of scripts we could leverage.
    Check Javascript.
    Check CSS.
    Check SQL.
</pre>

### CODE REVIEW

**Comment by rdoursenaud** [comment]
This is ~20 hours of hard work. Hope you'll find it useful!
Now on to the TODOs — after a bit of sleep ;)

File: dev/codesniffer/ruleset.xml [hunk 1]
```diff
@@ -3,11 +3,16 @@
 <ruleset name="Dolibarr">
 	<description>Dolibarr coding standard.</description>
 
-	<exclude-pattern>*/conf.php</exclude-pattern>
-	<exclude-pattern>*/includes/*</exclude-pattern>
-	<exclude-pattern>*/documents/*</exclude-pattern>
-	<exclude-pattern>*/dev/vagrant/*</exclude-pattern>
-	<exclude-pattern>*/custom/*</exclude-pattern>
+	<exclude-pattern>build/html</exclude-pattern>
+	<!--
```
Comment (eldy) [inline dev/codesniffer/ruleset.xml]: Why did not you keep pattern  _/documents/_  so it is not bugged ? Only dir is excluded

Comment (rdoursenaud) [inline dev/codesniffer/ruleset.xml]: As explained, any option containing the string "documents" is buggy. I don't know why…
There's no php or js file to check there anyway so it should not be a major issue.

Comment (rdoursenaud) [inline dev/codesniffer/ruleset.xml]: Oh crap. I figured it out. I have a documents folder in my path !
Pull request using relative path instead incoming.

**Comment by rdoursenaud** [comment]
Should be good to go as soon as the build is finished :)

**Pull request merged**